signer/src/network/libp2p/event_loop.rs

```diff
@@ -31,131 +29,182 @@ pub async fn run(ctx: &impl Context, swarm: Arc<Mutex<Swarm<SignerBehavior>>>) {
     let mut signal_rx = ctx.get_signal_receiver();
     let signal_tx = ctx.get_signal_sender();
 
-    loop {
-        tokio::select! {
-            // Handle termination signals from the application
-            _ = term.wait_for_shutdown() => {
-                tracing::info!("libp2p received a termination signal; stopping the libp2p swarm");
-                break;
-            },
-            // Handle signals from the application
-            Ok(SignerSignal::Command(cmd)) = signal_rx.recv() => {
-                match cmd {
-                    // Handle a request to publish a message to the P2P network.
-                    SignerCommand::P2PPublish(payload) => {
-                        let msg_id = payload.id();
+    // Here we create a future that listens for `P2PPublish` commands from the
+    // app signalling channel and pushes them into the outbound message queue.
+    // This queue is then polled by the `poll_swarm` event loop to publish the
+    // messages to the network.
+    let outbox = RwLock::new(Vec::<Msg>::new());
```

Comment:
nit: `Mutex` instead of `RwLock`?

Reply:
Must've been a leftover from an attempt at something else, fixed in [60ca5c9](https://github.com/stacks-network/sbtc/pull/728/commits/60ca5c9a8db297e70464204e6d9d6746dcdfb205)